yarn-project/p2p/src/testbench/p2p_client_testbench_worker.ts

```diff
@@ -104,6 +103,22 @@ process.on('message', async msg => {
         deps,
       );
 
+      // We disable message validation in the testbench
+      // In our case we only deal with txs
+      (client as any).p2pService.handleGossipedTx = async (msg: Message, msgId: string, source: PeerId) => {
+        const tx = Tx.fromBuffer(Buffer.from(msg.data));
+        (client as any).p2pService.node.services.pubsub.reportMessageValidationResult(
+          msgId,
+          source.toString(),
+          TopicValidatorResult.Accept,
+        );
+
+        const txHash = await tx.getTxHash();
+        const txHashString = txHash.toString();
+        logger.verbose(`Received tx ${txHashString} from external peer ${source.toString()}.`);
+        await txPool.addTxs([tx]);
+      };
```

Comment:
I feel like we've been abusing a lot of overwriting specific functions behind an `as any` (and I've been one of the main culprits). Maybe we should shoot for something like a `TestP2PClient`, which extends the main one, and has a flag for disabling tx validation? So we don't bypass typechecks so egregiously.